lib/Server.js

```diff
@@ -67,13 +65,24 @@ class Server {
 
     this.log = _log || createLogger(options);
 
+    // set serverMode default
     if (this.options.serverMode === undefined) {
       this.options.serverMode = 'sockjs';
     } else {
       this.log.warn(
         'serverMode is an experimental option, meaning its usage could potentially change without warning'
       );
     }
+    // set clientMode default
+    if (this.options.clientMode === undefined) {
+      this.options.clientMode = 'sockjs';
+    } else {
+      this.log.warn(
+        'clientMode is an experimental option, meaning its usage could potentially change without warning'
+      );
+    }
+
+    updateCompiler(this.compiler, this.options);
```

Comment:
Why we move this?

Reply:
@evilebottnawi I reorganized it to have the default setting for `serverMode` and `clientMode` together above that. `updateCompiler` uses `this.options.clientMode`, so the default setting for it must be set before `updateCompiler` is called. It can be switched back, I just thought it would be nice to have default setting for `serverMode` and `clientMode` right next to one another